src/DefaultBuilder/src/WebApplicationOptions.cs

```diff
@@ -60,6 +65,12 @@ internal void ApplyHostConfiguration(IConfigurationBuilder builder)
                 config[HostDefaults.ContentRootKey] = ContentRootPath;
             }
 
+            if (WebRootPath is not null)
```

Comment:
I feel like we should be applying this to the IWebHostBuilder via UseSetting. It still gets applied right?

Reply:
It gets applied when we run the default callbacks on the `BootstrapHostBuilder`. 

> I feel like we should be applying this to the IWebHostBuilder via UseSetting. 

Aren't we explicitly preventing it from being applied on the IWebHostBuilder with this change? Unless you're referring to a different `IWebHostBuilder` implementation than `ConfigureWebHostBuilder`

Reply:
> Aren't we explicitly preventing it from being applied on the IWebHostBuilder with this change? Unless you're referring to a different `IWebHostBuilder` implementation than `ConfigureWebHostBuilder`

I think @davidfowl is asking if we should call UseSetting() on the GenericWebHostBuilder we get in the _bootstrapHostBuilder.ConfigureWebHostDefaults() callback like we do in WebApplicationOptions.ApplyApplicationName().

Calling UseSetting() is normally equivalent to just setting host configuration, but we have to call UseSetting() for ApplicationName because GenericWebHostBuilder [uses it's own internal config in ExecuteHostingStartups()](https://github.com/dotnet/aspnetcore/blob/047dde71e5e5255a90ce5fefbe9ca6d82565994d/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs#L127) which depends on ApplicationName.

The internal config is chained to the host config which is where configuration is normally read from even in GenericWebHostBuilder, but ExecuteHostingStartups() runs before the host configuration is built. We'd have to do the same thing if we added PreventHostingStartup to WebApplicationOptions, but I don't think it's necessary for WebRootPath.

Reply:
I guess the other thing it could influence is IHostingStartups calling GetSetting(). But if we're concerned about that, we should be calling UseSetting for everything and we're not currently.

Reply:
> We'd have to do the same thing if we added PreventHostingStartup to WebApplicationOptions, but I don't think it's necessary for WebRootPath.

Thanks for the explainer! It's actually a bit of a gotcha that you need to know which options require special handling because they get treated differently in the `GenericWebHostBuilder` and which don't. I can observe `ApplicationName` and `PreventHostingStartup` from the code above but are there any other gotchas?